code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -261,43 +287,50 @@ GLOBAL_VAR_INIT(dynamic_forced_threat_level, -1)
 		if(istext(entry))
 			out += "[entry]<BR>"
 
-	out += "<B>Remaining threat/threat_level:</B> [threat]/[threat_level]"
+	out += "<B>Remaining threat/threat_level:</B> [mid_round_budget]/[threat_level]"
 
 	usr << browse(out.Join(), "window=threatlog;size=700x500")
 
 /// Generates the threat level using lorentz distribution and assigns peaceful_percentage.
 /datum/game_mode/dynamic/proc/generate_threat()
-	var/relative_threat = LORENTZ_DISTRIBUTION(GLOB.dynamic_curve_centre, GLOB.dynamic_curve_width)
-	threat_level = round(lorentz_to_threat(relative_threat), 0.1)
+	var/relative_threat = LORENTZ_DISTRIBUTION(threat_curve_centre, threat_curve_width)
+	threat_level = round(lorentz_to_amount(relative_threat), 0.1)
 
-	peaceful_percentage = round(LORENTZ_CUMULATIVE_DISTRIBUTION(relative_threat, GLOB.dynamic_curve_centre, GLOB.dynamic_curve_width), 0.01)*100
+	peaceful_percentage = round(LORENTZ_CUMULATIVE_DISTRIBUTION(relative_threat, threat_curve_centre, threat_curve_width), 0.01)*100
 
-	threat = threat_level
+/// Generates the midround and roundstart budgets
+/datum/game_mode/dynamic/proc/generate_budgets()
+	var/relative_round_start_budget_scale = LORENTZ_DISTRIBUTION(roundstart_split_curve_centre, roundstart_split_curve_width)
+	round_start_budget = round((lorentz_to_amount(relative_round_start_budget_scale) / 100) * threat_level, 0.1)
+	initial_round_start_budget = round_start_budget
+	mid_round_budget = threat_level - round_start_budget
 
 /datum/game_mode/dynamic/can_start()
 	message_admins("Dynamic mode parameters for the round:")
-	message_admins("Centre is [GLOB.dynamic_curve_centre], Width is [GLOB.dynamic_curve_width], Forced extended is [GLOB.dynamic_forced_extended ? "Enabled" : "Disabled"], No stacking is [GLOB.dynamic_no_stacking ? "Enabled" : "Disabled"].")
-	message_admins("Stacking limit is [GLOB.dynamic_stacking_limit], Classic secret is [GLOB.dynamic_classic_secret ? "Enabled" : "Disabled"].")
+	message_admins("Centre is [threat_curve_centre], Width is [threat_curve_width], Forced extended is [GLOB.dynamic_forced_extended ? "Enabled" : "Disabled"], No stacking is [GLOB.dynamic_no_stacking ? "Enabled" : "Disabled"].")
+	message_admins("Stacking limit is [GLOB.dynamic_stacking_limit].")
```

Comment:
These do not need to be admin messages, its just spammy and is available in game panel + logs if anyone needs to look it up. It cant be changed at this point anyway. Its been on my list forever.